job_executor/model/datastore.py

```diff
@@ -199,6 +199,9 @@ def bump_version(self, bump_manifesto: DatastoreVersion, description: str):
             [] if self.metadata_all_latest is None
             else [ds for ds in self.metadata_all_latest]
         )
+
+        local_storage.archive_draft_version(new_version)
```

Comment:
this needs to be moved above line 191, due to line 191 mutating the draft file we wish to archive:
```py
if not self.draft_version.validate_bump_manifesto(bump_manifesto):  # currently line 186
    raise VersioningException(
        'Invalid Bump: Changes were made to the datastore '
        'after bump was requested'
    )
local_storage.archive_draft_version(self.latest_version_number)  # here
release_updates, update_type = self.draft_version.release_pending() # currently line 191
```
We can use the "pre-bump" version number 👍🏻

Reply:
Ahh, good point